src/main/java/com/jvm_bloggers/core/social/twitter/TweetContentGenerator.java

```diff
@@ -0,0 +1,66 @@
+package com.jvm_bloggers.core.social.twitter;
+
+import com.jvm_bloggers.core.blogpost_redirect.LinkGenerator;
+import com.jvm_bloggers.entities.newsletter_issue.NewsletterIssue;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Random;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static lombok.AccessLevel.PACKAGE;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(access = PACKAGE)
+class TweetContentGenerator {
+
+    private static final int TWEET_MAX_LENGTH = 140;
+    private static final String MESSAGE_TEMPLATE =
+        "Nowy numer #%s już online - %s z postami m.in. %s, %s i %s #java #jvm";
+    private static final String SHORT_MESSAGE_TEMPLATE =
+        "Nowy numer #%s już online - %s z postami m.in. %s i %s #java #jvm";
+
+    private final LinkGenerator linkGenerator;
+    private final Random random = new Random();
+
+    public String generateTweetContent(NewsletterIssue issue) {
+        final List<String> personalTTs =
```

Comment:
6. Try to avoid unnecessary variables and use method references where possible, eg. 
```java
issue.getBlogPosts().stream()
    .map(BlogPost::getBlog)
    .filter(Blog::isPersonal)
    .map(Blog::getTwitter)
    .collect(toList());
```